src/Raven.Server/Utils/RavenWindowsServiceController.cs

```diff
@@ -0,0 +1,225 @@
+            }
+        }
+
+        private static void UninstallInternal(ServiceController serviceController)
+        {
+            try
+            {
+                if ((serviceController.Status == ServiceControllerStatus.Stopped || serviceController.Status == ServiceControllerStatus.StopPending) == false)
+                {
+                    StopInternal(serviceController);
+                }
+
+                new Win32ServiceManager().DeleteService(
+                    NormalizeServiceName(RavenWindowsService.WindowsServiceName));
+
+                Console.WriteLine($@"Successfully unregistered service {ServiceFullName}");
+            }
+            catch (Exception e) when (e.Message.Contains("does not exist"))
+            {
+                Console.WriteLine($@"Service {ServiceFullName} does not exist. No action taken.");
+            }
+        }
+
+        private static void Reinstall(ServiceController sc, List<string> serviceArgs)
+        {
+            StopInternal(sc);
+            UninstallInternal(sc);
+            InstallInternal(sc, serviceArgs);
+        }
+
+        private static void StopInternal(ServiceController serviceController)
+        {
+            if (!(serviceController.Status == ServiceControllerStatus.Stopped | serviceController.Status == ServiceControllerStatus.StopPending))
+            {
+                serviceController.Stop();
+                serviceController.WaitForStatus(ServiceControllerStatus.Stopped, TimeSpan.FromMilliseconds(1000));
+                Console.WriteLine($@"Successfully stopped service {ServiceFullName}");
+            }
+            else
+            {
+                Console.WriteLine($@"Service {ServiceFullName} is already stopped or stop is pending.");
+            }
+        }
+
+        public static void Stop()
+        {
+            using (var serviceController = GetServiceController())
+            {
+                StopInternal(serviceController);
+            }
+        }
+
+        public static void Start()
+        {
+            using (var serviceController = GetServiceController())
+            {
+                StartInternal(serviceController);
+            }
+        }
+
+        public static void Run(RavenConfiguration configuration)
+        {
+            var service = new RavenWindowsService(configuration);
+            var serviceHost = new Win32ServiceHost(service);
+            serviceHost.Run();
+        }
+
+        private static void StartInternal(ServiceController serviceController)
+        {
+            if (!(serviceController.Status == ServiceControllerStatus.StartPending | serviceController.Status == ServiceControllerStatus.Running))
+            {
+                serviceController.Start();
+                serviceController.WaitForStatus(ServiceControllerStatus.Running, TimeSpan.FromMilliseconds(1000));
+                Console.WriteLine($@"Successfully started service {ServiceFullName}");
```

Comment:
Can take more than 1 sec to start in some cases (recovery, frex).
Need to check if we started